piptools/utils.py

```diff
@@ -115,7 +115,16 @@ def format_requirement(
     if ireq.editable:
         line = f"-e {ireq.link.url}"
     elif is_url_requirement(ireq):
-        line = ireq.link.url
+        if not ireq.name:
+            line = ireq.link.url
+        elif ireq.link.url.startswith("file:./"):
```

Comment:
How `file:.` would work?

Reply:
This is to preserve an existing "hack" that allows us to specify a relative path, as mentioned [here](https://github.com/jazzband/pip-tools/issues/204#issuecomment-550051424).
Although, I can see this is not good enough: if the relative path does not start with `"./"`, such as `file:src/foo#egg=foo` then this does not work.
I'm going to update this.

Reply:
The logic has been updated, so I am resolving this conversation if you don't mind.